### PR TITLE
Added PR Template & Codeowners File

### PR DESCRIPTION
**Pull request by ilopezro**
Problem & Solution 
=======
In order to streamline our work, I have this PR template that will help us write better PR's. [This C5 article](https://blog.carbonfive.com/why-write-good-pull-requests/) does a really good job in explaining why good pull requests are crucial to software development.

I have also added the [`CODEOWNERS`](https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/about-code-owners) file that will automatically ping the @CruzHacks/2021-engineers whenever there is a PR ready for review.

### CODE REVIEW

**Review by tim-nguyen-cs — CHANGES_REQUESTED** [review]
Made a small comment on the verification section...

Something I also think would be a great addition is if we included a checklist to just validate some simple agreements our team think about. From a developer's perspective, I was thinking about something like this:

[] - If a UI Component, reviewed layout on desktop and mobile.
[] - Passes all build tests
[] - Rebased code to squash commits
[] - Is mergeable with `main` branch
...

Something like this that we can create throughout all our revisions will probably help reduce iteration time overall.

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+* Another great commit message
+* Something else I/we did
+
+Steps to Verify:
+----------------
+1. A setup step / beginning state
+2. What to do next
+3. Any other instructions
+4. Expected behavior
+5. Suggestions for testing
```
[the remark this answers is not in the document]
Comment (ilopezro) [reply .github/pull_request_template.md:25]: This section is basically so you can outline the steps we can take to reproduce your changes on our end. So if you implemented a navbar, these steps could look something like: 

1. Navigate to home page 
2. Verify that navbar is at the top & is sticky 
3. When clicking on `Contact Us`, it should navigate to the `/contact` end point 
4. Verify that navbar is still sticky on top

[the remark this answers is not in the document]
Comment (mocklee) [reply .github/pull_request_template.md:25]: so the verify section essentially creates a quick proof of user story completion?

**Comment by mocklee** [comment]
@tim-nguyen-cs A merge checklist would be a solid addition. I think once CI is integrated we'll easily be able to see if build tests pass, and any reviewer should definitely verify that the PR meets design specs. Squashing commits isn't something we've thought about before— but if the commit histories get messy, a rebase & squash could be useful with reverting https://thoughtbot.com/blog/git-interactive-rebase-squash-amend-rewriting-history

File: .github/pull_request_template.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (tim-nguyen-cs) [reply .github/pull_request_template.md:25]: Ahh, that makes sense - thanks for the clarification.

**Comment by tim-nguyen-cs** [comment]
Yeahh, that makes sense @mocklee  - although most of the work should be verified by reviewers as they review code, I think a checklist would help improve developers responsiveness when replying to issues. Say I made a PR but one of the tests failed, if we had a checklist, it could help me explicitly remember to check certain things instead of waiting until later.

**Comment by ilopezro** [comment]
Sometimes, tickets may be a little ambiguous as to how they should be accepted. Code review is one part of it – the second is making sure the UI is working. This section ensures that I can quickly checkout your branch, and follow the steps to verify that your UI component works as desired. CI/CD will run all the system and unit tests that need to be run. That's a whole separate process. We will be requiring CI to pass before being able to merge. As developers, we will tell you: "x feature does not work as desired" and maybe suggest code changes, but at the end of the day you're in charge of making the changes. The `Steps to Verify` section is so streamline the developers review process so we don't have to open up the Asana ticket and read through what the ticket is intended to do. @tim-nguyen-cs

**Comment by tim-nguyen-cs** [comment]
@ilopezro I'm understanding what you're saying, but I'm not sure that relates to the point I'm trying to make. Independent of anything related to the `Steps to Verify` section or anything having to do with Asana tickets, I just think a checklist would be another safeguard to improve development efficiency, reducing communication time between developers and reviewers as code is being reviewed if, say, developers "forgot" to do something.

Am I being clear?

**Comment by ilopezro** [comment]
@tim-nguyen-cs yeah that makes sense to me. The neat thing about GitHub is you can create draft PR's and ask for comments then. Once you open up a PR, we don't always expect it to be 100% clean code, but we do expect top quality code from our developers. Asana tickets are typically where you as the dev in charge should keep track of that. PR's should consist of code that you are confident is production ready. As reviewers, we will check our your code & make sure everything checks out. If there is something we see that needs fixing, we will request changes & let you know what needs to be changed.

**Review by tim-nguyen-cs — APPROVED** [review]

**Pull request merged**